docs/linker-attributes.md

```diff
@@ -0,0 +1,78 @@
+using System.Diagnostics.CodeAnalysis;
+
+class Program
+{
+    [DynamicDependency("Helper")]
+    public static void Main()
+    {
+        var helper = Assembly.GetExecutingAssembly().GetType("Program").GetMethod("Helper");
+        helper.Invoke(null, null);
+    }
+
+    public static void Helper()
+    {
+        Console.WriteLine("Dynamic call to helper");
+    }
+}
+```
+
+Without `DynamicDependency`, linking this assembly would remove `Helper`, causing a failure at runtime. The attribute ensures that `Helper` is kept.
+
+The attribute specifies the members to keep via a `string` or via `DynamicallyAccessedMemberTypes`. The type and assembly are either implicit in the attribute context, or explicitly specified in the attribute (by `Type`, or by `string`s for the type and assembly name).
+
+The type and member strings use the format described at https://github.com/dotnet/csharplang/blob/master/spec/documentation-comments.md#id-string-format, without the member prefix. The member string should not include the name of the declaring type, and may omit parameters to keep all members of the specified name. Some examples of the format follow:
+
+```csharp
+[DynamicDependency("Method()")]
+[DynamicDependency("Method(System,Boolean,System.String)")]
+[DynamicDependency("MethodOnDifferentType()", typeof(ContainingType))]
+[DynamicDependency("MemberName")]
+[DynamicDependency("MemberOnUnreferencedAssembly", "ContainingType", "UnreferencedAssembly")]
+[DynamicDependency("MemberName", "Namespace.ContainingType.NestedType", "Assembly")]
+
+// generics
+[DynamicDependency("GenericMethodName`1")]
+[DynamicDependency("GenericMethod``2(``0,``1)")]
+[DynamicDependency("MethodWithGenericParameterTypes(System.Collections.Generic.List{System.String})")]
+[DynamicDependency("MethodOnGenericType(`0)", "GenericType`1", "UnreferencedAssembly")]
+[DynamicDependency("MethodOnGenericType(`0)", typeof(GenericType<>))]
+```
+
+This attribute will most often be used in cases where a method contains reflection patterns that can not be analyzed even with the help of DynamicallyAccessedMembers. In these cases, the linker will warn about non-understood patterns in the method body. Applying DynamicDependencyAttribute to the method allows the dynamically referenced members to be kept. The warnings will still show up unless silenced by another mechanism (see UnconditianalSuppressMessage).
+
+## UnconditionalSuppressMessage
+
+Silences linker warnings originating from a method.
+
+TODO: provide examples
+
+In some cases, the linker may warn about unrecognized reflection access patterns which the developer knows to be safe. For example, the accessed members may be referenced elsewhere (such as by DynamicDependency), or they may have safe fallback behavior when the reflection targets are trimmed by the linker. This attribute can be used to suppress messages about such warnings. It can also suppress other linker warnings that are unrelated to the analysis.
```

Comment:
Don't use "safe" and/or "unsafe" - it has too much "security" feeling to it. Maybe "trim friendly" or "trim correct" and so on...